### PR TITLE
feat: RDCC-1661 Merge FPLA and AAC org identifier changes to master

### PR DESCRIPTION
**Pull request by shreedhar-hmcts**
### JIRA link (if applicable) ###
RDCC-1661

### Change description ###
feat: RDCC-1661 Merge FPLA and AAC org identifier changes to master

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/RetrieveOrganisationByOrgIdTest.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    @Test
+    public void rdcc117_ac1_pui_finance_manager_can_retrieve_organisation_by_orgIdentifier_for_external() {
+        Map<String, Object> response = professionalApiClient.retrievePbaAccountsForAnOrganisationExternal(HttpStatus.OK, generateBearerTokenForUser(puiFinanceManager));
+        assertThat(response.get("paymentAccount")).asList().hasSize(3);
+        responseValidate(response);
+    }
+
+    @Test
+    public void rdcc117_ac2_pui_organisation_manager_can_retrieve_organisation_by_orgIdentifier_for_external() {
+        Map<String, Object> response = professionalApiClient.retrievePbaAccountsForAnOrganisationExternal(HttpStatus.OK, generateBearerTokenForUser(puiOrgManager));
```
Comment (KOTLAPRA23) [inline src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/RetrieveOrganisationByOrgIdTest.java:73]: Can you please delete the three retrievePbaAccountsForAnOrganisationExternal functional test which already present in the RetrievePaymentAccountTest

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/SuperController.java [hunk 2]
```diff
@@ -134,7 +140,7 @@
             organisationCreationRequestValidator.validateCompanyNumber(organisationCreationRequest);
         }
 
-        if (StringUtils.isBlank(organisationCreationRequest.getSraRegulated())) {
+        if (isBlank(organisationCreationRequest.getSraRegulated())) {
             organisationCreationRequest.setSraRegulated(SRA_REGULATED_FALSE);
```
Comment (KOTLAPRA23) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/controller/SuperController.java:144]: two if conditions are same  i think we can remove one.
isBlank(organisationCreationRequest.getSraRegulated())

Comment (shreedhar-hmcts) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/controller/SuperController.java:144]: Sure can be removed. Its production code, can be refactored in next release after QA regression.

File: src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/RetrieveOrganisationByOrgIdTest.java (hunk 1 above)
Comment (shreedhar-hmcts) [reply src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/RetrieveOrganisationByOrgIdTest.java:73]: Yes can be removed in next release.

File: src/test/java/uk/gov/hmcts/reform/professionalapi/util/RefDataUtilTest.java [hunk 3]
```diff
@@ -311,19 +353,26 @@ public void test_shouldCreatePageableObjectWithDefaultPageSize() {
...
-
-        assertThat(response.equals("false")).isTrue();
+    public void test_getReturnRolesValue() {
+        assertThat(RefDataUtil.getReturnRolesValue(Boolean.valueOf("True"))).isEqualTo(true);
+        assertThat(RefDataUtil.getReturnRolesValue(Boolean.valueOf("true"))).isEqualTo(true);
+        assertThat(RefDataUtil.getReturnRolesValue(Boolean.valueOf("TRUE"))).isEqualTo(true);
+        assertThat(RefDataUtil.getReturnRolesValue(Boolean.valueOf("False"))).isEqualTo(false);
+        assertThat(RefDataUtil.getReturnRolesValue(Boolean.valueOf("false"))).isEqualTo(false);
+        assertThat(RefDataUtil.getReturnRolesValue(Boolean.valueOf("FALSE"))).isEqualTo(false);
     }
```
Comment (shashank-rastogi) [inline src/test/java/uk/gov/hmcts/reform/professionalapi/util/RefDataUtilTest.java:376]: why is this  test required? hv not we converted the query parameter to boolean ?

Comment (shreedhar-hmcts) [reply src/test/java/uk/gov/hmcts/reform/professionalapi/util/RefDataUtilTest.java:376]: yes , its correct , need to change parameter to boolean.

File: src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/RetrieveOrganisationByOrgIdTest.java (hunk 1 above)
Comment (shreedhar-hmcts) [reply src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/RetrieveOrganisationByOrgIdTest.java:73]: Created tech debt for this

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/SuperController.java (hunk 2 above)
Comment (shreedhar-hmcts) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/controller/SuperController.java:144]: Created tech debt for this

File: src/test/java/uk/gov/hmcts/reform/professionalapi/util/RefDataUtilTest.java (hunk 3 above)
Comment (shreedhar-hmcts) [reply src/test/java/uk/gov/hmcts/reform/professionalapi/util/RefDataUtilTest.java:376]: Created tech debt for this

Comment (shreedhar-hmcts) [reply src/test/java/uk/gov/hmcts/reform/professionalapi/util/RefDataUtilTest.java:376]: https://tools.hmcts.net/jira/browse/RDCC-1680

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/SuperController.java (hunk 2 above)
Comment (shreedhar-hmcts) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/controller/SuperController.java:144]: https://tools.hmcts.net/jira/browse/RDCC-1680

File: src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/RetrieveOrganisationByOrgIdTest.java (hunk 1 above)
Comment (shreedhar-hmcts) [reply src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/RetrieveOrganisationByOrgIdTest.java:73]: https://tools.hmcts.net/jira/browse/RDCC-1680

**Comment by shreedhar-hmcts** [comment]
Pre please tag created : v3.0.16 
Ready for merging into master

**Pull request merged**